Source/Scene/Scene.js

```diff
@@ -1124,9 +1124,20 @@ define([
         if (command._dirty) {
             command._dirty = false;
             
+            var context = scene._context;
+            var derivedCommands = command.derivedCommands;
+
+            if (defined(scene.shadowMap)) {
+                derivedCommands.shadows = scene.shadowMap.createDerivedCommands(command, context, derivedCommands.shadows);
+            }
```

Comment:
One potential downside here is an object won't be able to cast shadows unless it has been viewed by the scene camera first.